third_party/tsl/tsl/platform/cpu_info.cc

```diff
@@ -458,18 +464,50 @@ class CPUIDInfo {
     }
 #endif  // !PLATFORM_WINDOWS
   }
+  static void InitializeCPUFeature() {
+    // Initialize CPUIDInfo pointer.
+    if (cpuid != nullptr) return;
+
+    cpuid = new CPUIDInfo;
+    // Make sure CPUID registers are available before reading them.
+    if (!(getauxval(AT_HWCAP2) & ARM_COMPUTE_CPU_FEATURE_HWCAP2_BF16)) {
+      return;
+    }
+    const uint32_t hwcaps2 = getauxval(AT_HWCAP2);
+    cpuid->has_bf16 =
+        is_feature_supported(hwcaps2, ARM_COMPUTE_CPU_FEATURE_HWCAP2_BF16);
+  }
 
   int implementer() const { return implementer_; }
   int cpunum() const { return cpunum_; }
 
   static bool TestAarch64CPU(Aarch64CPU cpu) {
     InitCPUIDInfo();
+    // clang-format off
+
     switch (cpu) {
       case ARM_NEOVERSE_V1:
         return cpuid->is_arm_neoverse_v1_;
       default:
-        return 0;
+        return false;
     }
+    // clang-format on
+    return false;
+  }
+
+  static bool is_feature_supported(uint64_t features, uint64_t feature_mask) {
```

Comment:
Nit: According to Google's [C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Function_Names), we prefer camel case (`IsFeatureSupported`) for non-getter/setter member functions. 
```suggestion
  static bool IsFeatureSupported(uint64_t features, uint64_t feature_mask) {
```